### ISSUE TITLE
Request line for WCF methods

### ISSUE THREAD

**Issue by dodexahedron** [dodexahedron/SE-Community-Mod-API#4]
What would you like to see externally available, via WCF, from SESE?
Please post your suggestions here.

**Comment by dodexahedron** [comment]
I've started work on this.
I think an obvious first place to start is with the basics - starting and stopping the server.
I should have a release available pretty soon with that, since it shouldn't be so difficult to get that in there.
I may or may not have any time to actually get to it tonight, though.

Any other requests/suggestions for what to take care of after those?

**Comment by Devlah** [comment]
Some time ago, I tried to get SELiveMap working.  Clearly it uses WCF though, so it didn't work.  Also, quite a while back, I experimented with the GUI app that comes with SEServerExtender, only to find that it seemed to not really function.  The config file seems to have places to specify an address and port (defaulting to localhost) which led me to believe I may be able to use it to remote manage my server.  The version number was fairly old, the interface missing many features, and it crashed when trying to interact with the CubeGrids.  It sounds as if you're already well on the way to remedying this.

So, in short:

SEServerExtenderGUI (perhaps with some sort of login security feature) for remote server management.

SELiveMap would be nice for those of us who spend more time managing than playing.

**Comment by dodexahedron** [comment]
I have no plans, myself, to support SELiveMap at all, as I have no graphics programming experience that extends beyond 2D.

Neither one of those projects has been maintained at all since forking from the original SE-Community-Mod-API repository.
In fact, both it an the old SEServerExtenderGUI have been removed from my fork.  If someone else wants to revive SELiveMap, I definitely invite them to.

For remote management, the WCF methods that I'm implementing will be the framework for someone else to develop a front-end to, be it a desktop application, a web interface, a phone app, etc.
I'll of course be making a client for it of some sort, since I need that for testing, but it'll be relatively basic (like SESEGUI was).

**Comment by joshuataylor** [comment]
I would like to create a webapp for this, and the basic features for a prototype would be:
- Server management
  - Management of server (Start/stop)
  - See list of plugins used on the server
  - Probably look into create some sort of workaround to add new plugins
  - See workshop plugins currently used
- Player Management
  - See list of players, their name / steam ID / loc
  - Manage players (move them around, edit their inventories if possible)
- Entity Management
  - See a list of all entities in the game
  - Manage these entities

LOADS more ideas, but this is only the start.

**Comment by dodexahedron** [comment]
Good stuff.
Keep em coming!

-----Original Message-----

**Comment by dodexahedron** [comment]
https://github.com/dodexahedron/SE-Community-Mod-API/releases/tag/v1.71.05.114-alpha
This release has the ability to start and stop the server via WCF.
The default configuration is a completely unsecured web service on port 5353.
I'll be adding more of the functionality you mentioned very soon.

**Comment by joshuataylor** [comment]
Is this on by default? If so, and it is insecure for now, please please please make this a command line option to turn it on whilst this is the case.

People probably won't firewall their shit.

**Comment by dodexahedron** [comment]
New release forthcoming, with message-level security turned on by default.
It shouldn't let non-local users perform commands without providing valid local authentication credentials.

**Comment by dodexahedron** [comment]
Enjoy:
https://github.com/dodexahedron/SE-Community-Mod-API/releases/tag/v1.71.05.134-alpha

**Comment by dodexahedron** [comment]
That toy client I put in that release has a hard-coded configuration name, so it won't work so well for you for starting a server.... Totally forgot about that.
In any case, you can start using the SEComm library to develop WCF clients for the server, now.

**Comment by joshuataylor** [comment]
:+1: 

Awesome work @dodexahedron , really loving your work with this.

**Comment by dodexahedron** [comment]
I'm working on my next round of WCF goodness now.
I think the first couple of things I'll add will be a player list method, kick/ban methods, and chat methods (in no particular order, subject to change, batteries not included, while supplies last).

**Comment by dodexahedron** [comment]
Working on #8 right now

**Comment by dodexahedron** [comment]
Kick/ban/unban/list player methods are all finished in #8.
These features have been released in v1.72.08.161.
Enjoy.

**Comment by dodexahedron** [comment]
Closing this in favor of https://github.com/SEServerExtender/SEServerExtender/issues/1
